### PR TITLE
add grpc-node plugin with support for NPM registry

### PR DESCRIPTION
**Pull request by pkwarren**

### CODE REVIEW

File: plugins/grpc/node/v1.11.3/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# syntax=docker/dockerfile:1.4
+FROM node:18.12.1-bullseye AS build
```
Comment (pkwarren) [inline plugins/grpc/node/v1.11.3/Dockerfile:2]: Bumped this to latest Node LTS (from previous version).

File: plugins/grpc/node/v1.11.3/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+ARG TARGETARCH
+
+WORKDIR /build
+COPY --link package* .
+RUN /bin/bash <<'EOF'
+set -e
+# https://github.com/grpc/grpc-node/issues/1405#issuecomment-1195903603 - linux/arm64 binaries not available
```
Comment (pkwarren) [inline plugins/grpc/node/v1.11.3/Dockerfile:10]: Still unresolved - will require Github actions runners that support linux/arm64.

File: plugins/grpc/node/v1.11.3/buf.plugin.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+registry:
+  npm:
+    import_style: commonjs
+    deps:
+    - package: '@grpc/grpc-js'
+      version: '^1.7.0'
+    - package: 'google-protobuf'
+      version: '^3.21.2'
+  opts:
+  - grpc_js
```
Comment (pkwarren) [inline plugins/grpc/node/v1.11.3/buf.plugin.yaml:19]: There was actually a bug in the previous version: https://github.com/bufbuild/plugins/pull/153/files#diff-7c1a6156c206b8fc008eac7d05d4fea580630a18b83a9c63d3a1b8f62b539c3aL7

File: plugins/grpc/node/v1.11.3/buf.plugin.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+output_languages:
+  - javascript
+registry:
+  npm:
+    import_style: commonjs
+    deps:
+    - package: '@grpc/grpc-js'
+      version: '^1.7.0'
+    - package: 'google-protobuf'
+      version: '^3.21.2'
```
Comment (pkwarren) [inline plugins/grpc/node/v1.11.3/buf.plugin.yaml:17]: Added this dependency from protocolbuffers/js deps.

File: plugins/grpc/node/v1.11.3/buf.plugin.yaml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+version: v1
+name: buf.build/grpc/node
+plugin_version: v1.11.3
+source_url: https://github.com/grpc/grpc-node
+description: Generates NPM client and server stubs for the gRPC framework.
```
Comment (mfridman) [inline plugins/grpc/node/v1.11.3/buf.plugin.yaml:5]: Should this be Node instead of NPM?

**Review by mfridman — APPROVED** [review]

**Pull request merged**